lib/npg_pipeline/product/heron/upload/metadata_client.pm

```diff
@@ -0,0 +1,288 @@
+package npg_pipeline::product::heron::upload::metadata_client;
+
+use strict;
+use warnings;
+
+use Data::Dump qw(pp);
+use JSON;
+use LWP::UserAgent;
+use Moose;
+use MooseX::StrictConstructor;
+use URI;
+
+with qw[
+    WTSI::DNAP::Utilities::Loggable
+    WTSI::DNAP::Utilities::JSONCodec
+];
+
+our $VERSION = '0';
+
+our $JSON_CONTENT_TYPE = 'application/json';
+
+has 'username' =>
+    (isa           => 'Str',
+     is            => 'ro',
+     required      => 1,
+     documentation => 'The registered user name for the COG-Uk API account',);
+
+has 'token' =>
+    (isa           => 'Str',
+     is            => 'ro',
+     required      => 1,
+     documentation => 'The registered access token for the COG-UK API account',);
+
+has 'api_uri' =>
+  (isa           => 'URI',
+   is            => 'ro',
+   required      => 1,
+   default       => sub { return URI->new('https://localhost') },
```

Comment:
Would it be better not to have a default here?

Reply:
Default removed